client/app/calendar/tasks/hooks/useCreateTask.ts

```diff
@@ -0,0 +1,60 @@
+import { useQueryClient } from '@tanstack/react-query'
+import { useForm } from 'react-hook-form'
+import { useToast } from '@/_hooks'
+import { useAuthStore } from '@/_stores'
+import { isErrorWithMessage } from '@/_utils'
+import { useCreateTaskMutation } from './api'
+import { createTaskResolver, taskSchema } from '../schema'
+
+export const useCreateTask = () => {
+  const queryClient = useQueryClient()
+  const getHouseId = useAuthStore((state) => state.getHouseId)
+  const houseId = getHouseId()
+  const { toast } = useToast()
+
+  const form = useForm<taskSchema>({
+    resolver: createTaskResolver,
+    defaultValues: {
+      title: '',
+      date: undefined,
+      assigneeId: undefined,
+      note: '',
+    },
+  })
+
+  const { mutate, isPending } = useCreateTaskMutation()
+
+  const handleSuccess = () => {
+    // TODO: Need to invalidate queryKey that is related to tasks
+    queryClient.invalidateQueries({ queryKey: ['todos'] })
+    form.reset()
+    toast({ variant: 'success', title: 'Successfully created a task!' })
+  }
```

Comment:
I think after you successfully create the task, it's better to redirect to the tasks list!

Reply:
Or maybe calendar is better? because you can only create tasks from calenday page and see all the tasks.
But In the main page, you can only see tasks for one week.

Reply:
Yes sorry tasks list means calendar!